src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -8468,6 +8468,42 @@ namespace SurfaceGeometry {
         }
     }
 
+    void CheckWindowShadingControlSimilarForWindow(bool& ErrorsFound)
+    {
+        // J.Glazer 2020 - for each window check if all window shading controls on list are the same except for name, schedule name, construction, and material
+        for (auto theSurf : Surface) {
+            if (theSurf.HasShadeControl) {
+                if (theSurf.windowShadingControlList.size() > 1) {
+                    int firstWindowShadingControl = theSurf.windowShadingControlList.front();
+                    for (auto wsc = std::next(theSurf.windowShadingControlList.begin()); wsc != theSurf.windowShadingControlList.end(); ++wsc) {
+                        if (!isWindowShadingControlSimilar(firstWindowShadingControl, *wsc)) {
+                            ErrorsFound = true;
+                            ShowSevereError("CheckWindowShadingControlSimilarForWindow: Fenestration surface named \"" + theSurf.Name +
+                                        "\" has multiple WindowShadingContols that are not similar.");
+                            ShowContinueError("for: \"" + WindowShadingControl(firstWindowShadingControl).Name + " and: " + WindowShadingControl(*wsc).Name); 
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    bool isWindowShadingControlSimilar(int a, int b)
```

Comment:
@mjwitte so this is the routine that checks that various windowshadingcontrol object fields match. What do you think should be checked? I was concerned for unintended side affects of allowing these to be different so that is why I made it fairly restrictive, what do you think needs to be the same between window shading controls for a given window?

Reply:
@JasonGlazer I'm not sure anything needs to be the same except the matching surface, unless there are some precalculations that are done based on the specific shading control type. And, I think we're still restricting the surfaces to all be in the same zone, so that would match also. Along those lines, I suppose if Multiple Surface Control Type = Group, then the list of surfaces should match? Or maybe just a warning that one control may override the other for an entire group?